tally_ho/apps/tally/views/super_admin.py

```diff
@@ -360,6 +391,99 @@ def get(self, *args, **kwargs):
+
+    def get(self, *args, **kwargs):
+        tally_id = kwargs['tally_id']
+
+        return self.render_to_response(self.get_context_data(
+            duplicate_results=get_result_form_with_duplicate_results(
+                tally_id=tally_id),
+            tally_id=tally_id))
+
+
+class DuplicateResultFormView(LoginRequiredMixin,
+                              mixins.GroupRequiredMixin,
+                              mixins.TallyAccessMixin,
+                              SuccessMessageMixin,
+                              TemplateView):
+    group_required = groups.SUPER_ADMINISTRATOR
+    template_name = "super_admin/duplicate_result_form.html"
+    success_url = "duplicate-result-tracking"
+
+    def get(self, *args, **kwargs):
+        tally_id = kwargs['tally_id']
+        barcode = kwargs['barcode']
+        ballot_id = kwargs['ballot_id']
+        result_form = ResultForm.objects.get(barcode=barcode)
+        results = Result.objects.filter(result_form=result_form.id)
+
+        return self.render_to_response(self.get_context_data(
+            results_form_duplicates=get_result_form_with_duplicate_results(
+                ballot=ballot_id,
+                tally_id=tally_id),
+            results=results,
+            tally_id=tally_id,
+            ballot_id=ballot_id,
+            header_text="Form " + str(barcode)))
+
+    def post(self, *args, **kwargs):
+        tally_id = kwargs['tally_id']
+        ballot_id = kwargs['ballot_id']
+        post_data = self.request.POST
+        results_form_duplicates = get_result_form_with_duplicate_results(
+            ballot=ballot_id,
+            tally_id=tally_id)
+
+        if 'duplicate_reviewed' in post_data:
+            for results_form_duplicate in results_form_duplicates:
+                results_form_duplicate.duplicate_reviewed = True
+                results_form_duplicate.save()
+
+            self.success_message = _(
+                u"Successfully marked forms as duplicate reviewed")
+
+            messages.add_message(
+                self.request, messages.INFO, self.success_message)
+
+            return redirect(self.success_url, tally_id=tally_id)
+        elif 'send_clearance' in post_data:
+            pk = post_data.get('result_form')
+            result_form = get_object_or_404(
+                ResultForm, pk=pk, tally__id=tally_id)
+            result_form.duplicate_reviewed = True
+            result_form.form_state = FormState.CLEARANCE
+            result_form.save()
+
+            self.success_message = _(u"Form successfully sent to clearance")
+
+            messages.add_message(
+                self.request, messages.INFO, self.success_message)
+
+            return redirect(self.success_url, tally_id=tally_id)
+        elif 'send_all_clearance' in post_data:
+            for results_form_duplicate in results_form_duplicates:
+                if results_form_duplicate.form_state != FormState.ARCHIVED:
```

Comment:
why are we checking here but not above in the block starting line 455?

think we should be consistent and not check here or also check there, but we'd need to update the message there if we do check there

Reply:
Because the `send_clearance` button is only displayed for non archived form, check this [line](https://github.com/onaio/tally-ho/blob/79f5a33e376a8883eb0a66202c837c106be186ea/tally_ho/apps/tally/templates/super_admin/duplicate_result_form.html#L14) while the `send_all_clearance` to clearance button is always displayed.

Probably, we should change that and always display the send to clearance button for each form and display an error message when it fails. No?

Reply:
We can display the button but have it disabled, but we should still check on the server side since someone can custom craft a request with the ID of the disabled form